tests/unit/verticals/secret/output/test_json_output.py

```diff
@@ -273,7 +371,72 @@ def test_json_output(client, cache, name, input_patch, expected_exit_code):
                 for incident in result["incidents"]
                 for occurrence in incident["occurrences"]
             ]
-            check_occurrences_indices(occurrences, input_patch)
+            check_occurrences_indices(occurrences, MATCH_INDICES_FOR_PATCH[input_patch])
+
+        # all ignore sha should be in the output
+        assert all(
+            ignore_sha in json_flat_results
+            for result in results.results
+            for ignore_sha in group_policy_breaks_by_ignore_sha(
+                result.scan.policy_breaks
+            )
+        )
+
+
+@pytest.mark.parametrize(
+    "name, file_content, expected_exit_code",
+    [
+        ("multiline_secret", _MULTILINE_SECRET_FILE, 1),
+        ("single_line_secret", _SINGLE_LINE_SECRET_FILE, 1),
+        ("one_line_and_multiline_secrets", _ONE_LINE_AND_MULTILINE_FILE, 1),
+    ],
+    ids=[
+        "_MULTILINE_SECRET_FILE",
+        "_SINGLE_LINE_SECRET_FILE",
+        "_ONE_LINE_AND_MULTILINE_FILE",
+    ],
+)
+def test_json_output_for_files(
```

Comment:
I think we should be able to avoid the duplication between `test_json_output_for_patches()` and this function by merging `MATCH_INDICES_FOR_FILES` and `MATCH_INDICES_FOR_PATCH` and adding some logic to determine if the content is a file or a patch (either via a boolean argument or by looking at the start of the content)

Reply:
Added boolean `is_patch` on the test cases :+1: